locations/spiders/pick_n_pay.py

```diff
@@ -0,0 +1,49 @@
+import scrapy
+
+from locations.dict_parser import DictParser
+from locations.hours import DAYS, OpeningHours
+
+
+class PickNPaySpider(scrapy.Spider):
+    name = "pick_n_pay"
+    item_attributes = {"brand": "Pick n Pay", "brand_wikidata": "Q7190735"}
+    start_urls = ["https://api.pnp.co.za/stores/v2"]
+
+    def parse(self, response, **kwargs):
+        for store in response.json():
+            store["id"] = store.pop("storeId")
+            address = store["storeAddress"]
+            store["housenumber"] = address.get("streetNo")
+            store["street"] = address.get("street")
+            store["suburb"] = address.get("suburb")
+            store["city"] = address.get("city")
+            province = address.get("province") or {}  # can be JSON null
+            store["province"] = province.get("name")
+            store["country"] = address.get("country")
+            store["postcode"] = address.get("postalCode")
+            item = DictParser.parse(store)
+
+            oh = OpeningHours()
+            for day in store.get("tradingHours", []):
+                # dayId is 1 to 8, 1 being Monday
+                # dayId of 8 is only present on some 24/7 stores so we ignore it
+                if day["dayId"] < 8:
+                    oh.add_range(DAYS[day["dayId"] - 1], day["openTime"], day["closeTime"])
+            item["opening_hours"] = oh
+
+            if store["storeType"] == "CLOTHING":
+                item.update({"brand": "Pick n Pay Clothing", "brand_wikidata": "Q122964352"})
+            elif store["storeType"] == "LIQUOR":
+                item.update({"brand": "Pick n Pay Liquor", "brand_wikidata": "Q122764458"})
+            elif store["storeType"] == "EXPRESS":
+                item.update({"brand": "Pick n Pay Express", "brand_wikidata": "Q122764443"})
+            elif store["storeType"] in ("BOXER", "BOXER PUNCH", "BOXER SUPERSTOR"):
+                item.update({"brand": "Boxer", "brand_wikidata": "Q116586275"})
+            elif store["storeType"] == "BOXER LIQUOR":
+                item.update({"brand": "Boxer Liquors", "brand_wikidata": "Q122766666"})
+            elif store["storeType"] == "BOXER BUILD":
+                item.update({"brand": "Boxer Build", "brand_wikidata": "Q122766671"})
+            # Unhandled: "", "BPFORECOURT", "DAILY", "DC", "FAMILY", "HYPER", "LOCAL", "MARKET",
```

Comment:
Hyper is listed as `Pick n Pay Hypermarket` with 21 stores per page 11 of https://www.picknpayinvestor.co.za/downloads/annual-report/2023/integrated-annual-report-2023-spreads.pdf

I suggest that this sub-brand could be added as a new Wikidata item?

Reply:
You're right, that Wikidata item appears to be for a long defunct attempt to roll out Pick 'n Pay Hypermarket to Australia per evidence at https://www.reddit.com/r/brisbane/comments/m7l2yi/remember_when_the_pick_n_pay_hypermarket_opened/

I assume instead of repurposing that item for the current Pick n Pay Hypermarket chain in South Africa, that a new Wikidata item be created for the 21 stores.